### ISSUE TITLE
Check for Microsoft.AspNetCore.App failing if dotnet core is not installed | NetCoreCheck

### ISSUE THREAD

**Issue by Haripc525** [dotnet/deployment-tools#281]
Check for Microsoft.AspNetCore.App failing if dotnet core is not installed | NetCoreCheck

### Description

- When trying to check if aspdotnet is installed using netcorecheck.exe it was failing if dotnet core is not installed.

Please provide a solution to solve this issue.

**Comment by NikolaMilosavljevic** [comment]
[Triage] Please investigate.

**Comment by hc-codes** [comment]
Hi @MSLukeWest, could you give an update on this issue?

**Comment by MSLukeWest** [comment]
@hc-codes A few questions:
1) To be clear, are you saying you have the ASP.NET runtime installed and netcorecheck.exe is always failing?  Note that if you have no .NET runtime installed (Core, Desktop, or ASP.NET) a failure is expected.
2) What error code is returned when the failure happens (in cmd prompt "echo %errorlevel%")?  And what's in the log (found under %temp%, name starts with "dd_NetCoreCheck_")?
3) NetCoreCheck writes a temp file next to its current location, so make sure you either have admin privileges or it's running from a non-protected location.
4) Make sure you're running the correct platform version of NetCoreCheck.  For example, if you run the x86 version on a machine that only has x64 runtimes installed it will always fail.

**Comment by hc-codes** [comment]
1. Yes ASP.NET is installed and check is failing. 
2. Error Code: 12289
                    [02/22/24,01:23:08] ============= NetCoreCheck Start ===============
                    [02/22/24,01:23:08] get_hostfxr_path failed: '-2147450749'
                    [02/22/24,01:23:08] =============  NetCoreCheck End  ===============
3. Yes both are satisfied
4. Platforms are also correct

Command used: NETCorecheck.exe --runtimename Microsoft.AspNetCore.App --runtimeversion 6.0.4

**Comment by hc-codes** [comment]
@MSLukeWest Yes its correct if dotnet core is not installed netcorechecker is failing to execute and it seems like its a prerequisite for netcorechecker which shouldn't be the case

**Comment by hc-codes** [comment]
@MSLukeWest Do you have any updates regarding this issue?

**Comment by MSLukeWest** [comment]
> [@MSLukeWest](https://github.com/MSLukeWest) Do you have any updates regarding this issue?

There's currently no ETA for adding support for detecting ASP.NET.
